plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * DataSynchronizer class file.
+ */
+
+namespace Automattic\WooCommerce\Internal\DataStores\Orders;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * This class handles the data migration/synchronization for the custom orders table. Its responsibilites are:
+ *
+ * - Performing the initial table creation and filling (triggered by initiate_regeneration)
+ * - Syncrhonizing changes between the custom orders table and the posts table whenever changes in orders happen.
+ */
+class DataSynchronizer {
+
+	const CUSTOM_ORDERS_TABLE_DATA_REGENERATION_IN_PROGRESS = 'woocommerce_custom_orders_table_data_regeneration_in_progress';
+	const CUSTOM_ORDERS_TABLE_DATA_REGENERATION_DONE_COUNT  = 'woocommerce_custom_orders_table_data_regeneration_done_count';
```

Comment:
Since these are class constants, they don't need such verbose names do they? Would `IN_PROGRESS` and `DONE_COUNT` suffice?

What do you think about adding `_OPTION` or similar as a suffix to indicate these are just string option names?

Reply:
Hum... yes and no. `IN_PROGRESS` is too generic (what's exactly the thing that is in progress?), but maybe it can be shortened to `DATA_REGENERATION_IN_PROGRESS`. And yes, adding `_OPTION` makes sense.

Anyway given the discussion that have been ongoing since this pull request was created, it seems that all of this will have to change anyway (we are going towards the concept of "sync" more than regeneration or migration).